docs/adr/0002-delivery-channel-creation-and-integration.md

```diff
@@ -0,0 +1,71 @@
+# Tenant Configuration and Routing of Notifications in the Marain.UserNotifications service
+
+## Table of definitions
+
+| Word               | Definition                                                                                    |
+| ------------------ | --------------------------------------------------------------------------------------------- |
+| Consumer           | A service that uses the Marain.UserNotifications service                                      |
+| User               | The individual who is using an application that consumes the Marain.UserNotifications service |
+| Delivery Channel   | A third party service used to send a notification. Eg. Twilio, Airship, Sendgrid etc.         |
+| Communication Type | The mechanism of the delivery of a notification eg. Sms, Email, WebPush                       |
+
+## Status
+
+Proposed
+
+## Context
+
+Delivering notifications through different communication types (email, sms, web push notifications etc) is necessary in an extendable notifications service. Organizations should be able to choose which delivery channels (twilio, sendgrid, airship etc) they want to use for the associated communication type(s) as these said delivery channel could have different features or costs. This allows a consumer of `Marain.UserNotifications` to optimise their costs by determining the specific delivery channel for each communication type.
+
```

Comment:
I think we need to make it clear that there are already two separate APIs: the management API and the thing currently called the `ApiDeliveryChannelService` (a name which arguably becomes confusing once we introduce multiple channels).